src/util/mpsc.rs

```diff
@@ -370,42 +418,30 @@ mod tests {
         assert_eq!(rx.recv(), Ok(10));
         let elapsed = timer.elapsed();
         assert!(elapsed >= Duration::from_millis(100), "{:?}", elapsed);
-    }
 
-    #[test]
-    fn test_notify() {
-        let (tx1, rx1) = super::unbounded();
-        let (tx2, rx2) = mpsc::unbounded();
-        let mut rx2 = rx2.wait();
-        let j = thread::spawn(move || {
-            rx1.map_err(|_| ())
-                .forward(tx2.sink_map_err(|_| ()))
-                .wait()
-                .unwrap();
-        });
-        tx1.send(2).unwrap();
-        assert_eq!(rx2.next().unwrap(), Ok(2));
-        for i in 0..100 {
-            tx1.send(i).unwrap();
-        }
-        for i in 0..100 {
-            assert_eq!(rx2.next().unwrap(), Ok(i));
-        }
-        thread::spawn(move || {
-            for i in 100..10000 {
-                tx1.send(i).unwrap();
-            }
-        });
-        for i in 100..10000 {
-            assert_eq!(rx2.next().unwrap(), Ok(i));
-        }
-        // j should automatically stop when tx1 is dropped.
-        j.join().unwrap();
+        let (tx, rx) = super::unbounded::<u64>();
+        assert!(tx.is_empty());
+        assert!(tx.is_sender_connected());
+        assert_eq!(tx.len(), 0);
+        assert!(rx.is_empty());
+        assert_eq!(rx.len(), 0);
+        tx.send(2).unwrap();
+        tx.send(3).unwrap();
+        assert_eq!(tx.len(), 2);
+        assert_eq!(rx.len(), 2);
+        tx.close_sender();
+        assert_eq!(tx.send(3), Err(SendError(3)));
+        assert_eq!(tx.try_send(3), Err(TrySendError::Disconnected(3)));
+        assert!(!tx.is_sender_connected());
+        assert_eq!(rx.try_recv(), Ok(2));
+        assert_eq!(rx.recv(), Ok(3));
+        assert_eq!(rx.try_recv(), Err(TryRecvError::Empty));
+        // assert_eq!(rx.recv(), Err(RecvError));
```

Comment:
Uncomment?

Reply:
It will be uncommented when full close is implemented.